modules/external-api/src/main/java/org/opencastproject/external/endpoint/EventsEndpoint.java

```diff
@@ -1704,17 +1704,22 @@ public Response getEventScheduling(@HeaderParam("Accept") String acceptHeader, @
   @Path("{eventId}/scheduling")
   @Produces({ ApiMediaType.JSON, ApiMediaType.VERSION_1_1_0, ApiMediaType.VERSION_1_2_0 })
   @RestQuery(name = "updateeventscheduling", description = "Update an event's scheduling information.", returnDescription = "", pathParameters = {
-      @RestParameter(name = "eventId", description = "The event id", isRequired = true, type = STRING) }, restParameters = {
-      @RestParameter(name = "scheduling", isRequired = true, description = "Scheduling Information", type = STRING) }, reponses = {
+      @RestParameter(name = "eventId", description = "The event id", isRequired = true, type = Type.STRING) }, restParameters = {
+      @RestParameter(name = "scheduling", isRequired = true, description = "Scheduling Information", type = Type.STRING),
+      @RestParameter(name = "allowConflict", description = "Allow conflicts when updating scheduling", defaultValue = "false", isRequired = false, type = Type.BOOLEAN) }, reponses = {
       @RestResponse(description = "The  scheduling information for the specified event is updated.", responseCode = HttpServletResponse.SC_NO_CONTENT),
       @RestResponse(description = "The specified event has no scheduling information to update.", responseCode = HttpServletResponse.SC_NOT_ACCEPTABLE),
       @RestResponse(description = "The scheduling information could not be updated due to a conflict.", responseCode = HttpServletResponse.SC_CONFLICT),
       @RestResponse(description = "The specified event does not exist.", responseCode = HttpServletResponse.SC_NOT_FOUND) })
   public Response updateEventScheduling(@HeaderParam("Accept") String acceptHeader, @PathParam("eventId") String id,
-                                 @FormParam("scheduling") String scheduling) throws Exception {
+                                 @FormParam("scheduling") String scheduling,
+                                 @FormParam("allowConflict") boolean allowConflict) throws Exception {
     final ApiVersion requestedVersion = ApiMediaType.parse(acceptHeader).getVersion();
     final Opt<Event> event = indexService.getEvent(id, externalIndex);
 
+    if (!requestedVersion.isSmallerThan(ApiVersion.VERSION_1_2_0)) {
```

Comment:
This should be

`if (requestedVersion.isSmallerThan(ApiVersion.VERSION_1_2_0)) { [...]`

shouldn't it?

Reply:
Is it necessary to do this at all? It's optional and the default is false anyway. Seems like this would override the value if you requested an API version < 1.2.0 and you provided a true value here, it would get changed to false. Doesn't seem like a necessary thing to enforce, unless I'm missing something.

Reply:
Without that, the older versions would essentially support this new parameter, too. So strictly spoken: It is necessary.

Reply:
I think it's potentially confusing to silently change the value like that. If we really want to be strict about not accepting a parameter in an earlier API version that was added in a later version, then we should return a Bad Request rather than accept it and ignore it.

Reply:
@smarquard AFAIK, form parameters that are not specified in the method signature will simply be ignore, i.e. if you pass an unsupported form parameter, no request would respond with "bad request".  Setting the value to false therefore makes sense as it prevents the value to be set by API versions not supposed to support the option.

Reply:
Yes, it should be
`if (requestedVersion.isSmallerThan(ApiVersion.VERSION_1_2_0)) { [...]`

Corrected in my code.

I think that if it is not explicitly set to version 1.2.0 or above (by default it takes the newest version anyway) this should be set to false so that it reacts like the parameter doesn't exist for older versions.

Even if it is set to false by default which "Due to MH-13416 REST docs defaultValue not correctly set for boolean parameters..." is uncertain at this point.

Reply:
@TurRil The @RestParameters annotations are Opencast-specific. The only thing they are used for is the dynamic creation of the REST docs page. The only thing that "defaultValue = "false"" does it to set the default value in the REST docs page - this won't affect the actual REST call (except when performed by the REST docs page).

What does make sense, though:

Change:

` @FormParam("allowConflict") boolean allowConflict`

to 

` @FormParam("allowConflict") @DefaultValue("false") boolean allowConflict`

@FormParam and @DefaultValue are Java annotations used by JAX-RS.

Reply:
Thanks - added it.